src/Search.py

```diff
@@ -16,7 +16,7 @@ def a_star(self, start, goal):
         priority_queue = PriorityQueue()
         visited = set()
         # set up heuristic
-        self.heuristic.setup(start, goal)
+        self.heuristic.setup(self, start, goal)
```

Comment:
I think it has to do with how heuristic is passed in. Changing line 12 from `self.heuristic = heuristic` to `self.heuristic = heuristic()` might fix it if the class itself is being passed in instead of an instance.